src/scenes/GameScene.js

```diff
@@ -14,61 +14,88 @@ export default class Game extends Phaser.Scene {
     this.load.image("tileset", "./assets/tileset/tileset.png");
     this.load.tilemapTiledJSON("map", "./assets/map/map.json");
 
-    this.load.spritesheet("player", "./assets/character/characters.png", {
+    this.load.spritesheet("characters", "./assets/character/characters.png", {
       frameWidth: 52,
       frameHeight: 72,
     });
   }
 
   create() {
-    const tilemap = this.make.tilemap({ key: "map" });
-    const tileset = tilemap.addTilesetImage("tileset", "tileset");
+    this.tilemap = this.make.tilemap({ key: "map" });
+    const tileset = this.tilemap.addTilesetImage("tileset", "tileset");
 
-    tilemap.createLayer("Ground", tileset);
-    tilemap.createLayer("Collisions", tileset);
-    tilemap.createLayer("Above", tileset);
-
-    const playerSprite = this.add.sprite(0, 0, "player");
-    this.cameras.main.startFollow(playerSprite);
-
-    const gridMovementConfig = {
-      characters: [
-        {
-          id: "player",
-          sprite: playerSprite,
-          walkingAnimationMapping: 6,
-          startPosition: new Phaser.Math.Vector2(8, 8),
-        },
-      ],
-    };
-
-    this.gridMovementPlugin.create(tilemap, gridMovementConfig);
+    const groundLayer = this.tilemap.createLayer("Ground", tileset);
+    this.tilemap.createLayer("Collisions", tileset);
+    this.tilemap.createLayer("Above", tileset);
 
     this.socket = io();
-    this.stateStatus = null;
-    this.players = {};
-
-    this.socket.on("newPlayer", (player) => {
-      const id = player.playerId;
-      this.playerList.addPlayer(id);
-      this.displayServerMessage(`New player connected! ${id}`);
-      this.players[id] = {
-        playerId: id,
-      };
+    this.mainPlayer = null;
+    this.mainPlayerName = null;
+    this.players = [];
+
+    this.cameras.main.setBounds(
+      0,
+      0,
+      this.tilemap.widthInPixels,
+      this.tilemap.heightInPixels
+    );
+
+    this.socket.on("newPlayer", (newPlayer) => {
+      this.displayServerMessage(`New player connected! ${newPlayer.id}`);
     });
     this.socket.on("playerDisconnected", (id) => {
-      this.playerList.removePlayer(id);
+      // this.playerList.removePlayer(id);
       this.displayServerMessage(`Player has left: ${id}`);
-      delete this.players[id];
+      // delete this.players[id];
     });
-    this.socket.on("currentPlayers", (players) => {
+    this.socket.on("currentPlayers", (players, socketId) => {
```

Comment:
Using this message for establishing `mainPlayer` is misleading, we should use `newPlayer` for this instead.